### PR TITLE
Fix readtimeout error not throwing

### PR DESCRIPTION
**Pull request by atrakh**

### CODE REVIEW

File: src/LaunchDarkly.EventSource/EventSourceService.cs [hunk 1]
```diff
@@ -92,7 +92,7 @@ public EventSourceService(Configuration configuration)
                     {
                         using (var reader = new StreamReader(stream))
                         {
-                            while (!cancellationToken.IsCancellationRequested && !reader.EndOfStream)
+                            while (!cancellationToken.IsCancellationRequested)
```
Comment (arun251) [inline src/LaunchDarkly.EventSource/EventSourceService.cs]: Why don't we want `!reader.EndOfStream`?

Comment (atrakh) [reply src/LaunchDarkly.EventSource/EventSourceService.cs]: Strangely, accessing `reader.EndOfStream` causes `readTimeoutTask` to never resolve. Even a simple logging statement which accesses `reader.EndOfStream` reproduces this behavior.

**Comment by atrakh** [comment]
Tests were failing because they rely on EventSource to close when the end of the stream was reached. This is not the functionality we want for our SDK. So, adding a configurable `closeOnEndOfStream` property allows for tests to pass, and for read timeout to work.

Re-review?

File: src/LaunchDarkly.EventSource/EventSourceService.cs [hunk 2]
```diff
@@ -107,13 +107,22 @@ public EventSourceService(Configuration configuration)
...
+                                }
                             }
                         }
                     }
 
                     OnConnectionClosed();
                 }
             }
+            catch (HttpRequestException e) {
+                OnConnectionClosed();
```
Comment (ryanfollmer) [inline src/LaunchDarkly.EventSource/EventSourceService.cs]: You shouldn't need to call the OnConnectionClosed method here. The calling code handles basic Exception types and will call Connection Close - see the catch block in EventSource.cs in the ConnectToEventSourceAsync method: `CloseAndRaiseError(e);`

Comment (atrakh) [reply src/LaunchDarkly.EventSource/EventSourceService.cs]: I believe `OnConnectionClosed` should be the method passed into EventSource @ initialization time. EventSource.cs does close the client on error, but does not call `OnConnectionClosed`

Comment (ryanfollmer) [reply src/LaunchDarkly.EventSource/EventSourceService.cs]: EventSource.cs doesn't need to call OnConnectionClosed, rather the EventSourceService.cs calls this on the EventSource.cs. In the event of an error, EventSource itself will set the Ready State to Closed.

**Pull request merged**